### DISCUSSION TITLE
Batch formatting

### DISCUSSION THREAD

**Discussion by leana8959** [mfussenegger/nvim-jdtls#576] [Q&A] [answered]
Hello,

I would like to know if there's a way to batch format all files of the entire project. I know that there's google's formatter, but it seems to use a different set of rules than this language server and I would end up with two distinct formatting style...

Thanks for putting the effort into this project that I use regularly, and for reading my question !

**Comment by mfussenegger** [comment, marked as answer]
There is no "format_all", but the regular (neo)vim approach to apply a function over many files is to use one of `:cdo`, `:bufdo` or `:argdo`. Check out their help. E.g. if using the arglist you'd use something like `:argdo lua vim.lsp.buf.format()`

**Reply by leana8959** [reply]
I see, thank you for the answer !
